src/html/mod.rs

```diff
@@ -141,12 +143,20 @@ pub struct HtmlText {
 impl HtmlText {
     /// Creates a new [HtmlText] from the given string.
     pub fn from_str(value: &str) -> HtmlText {
-        let text = unescape_characters(value);
```

Comment:
`unescape_characters` used to be called when creating a new `HtmlText`. Now it is only called when getting the display text for a given `HtmlText`, so if that `HtmlText` never needs to display its text, the slow method will not be called.